src/demo-app/table/table-demo.html

```diff
@@ -172,6 +172,61 @@ <h3>MatTable Example</h3>
   </mat-paginator>
 </div>
 
+<h3>MatTable with First/Last Buttons Example</h3>
+
+<div class="demo-table-container demo-mat-table-example mat-elevation-z4">
+
+  <table-header-demo (shiftColumns)="displayedColumns.push(displayedColumns.shift())"
+                     (toggleColorColumn)="toggleColorColumn()">
+  </table-header-demo>
+
+  <mat-table [dataSource]="dataSource" [trackBy]="userTrackBy">
```

Comment:
Since the data source for this table doesn't update with the paginator, we can go ahead and just remove it. Maybe replace it with a JSON text of the latest paginator output?

Reply:
![image](https://user-images.githubusercontent.com/22766134/35471759-c9bb193e-0359-11e8-9a7c-5a4b3daae6d8.png)

How about this?